### PR TITLE
Migrate Imbu from numenta-apps to nupic.research

### PR DESCRIPTION
**Pull request by oxtopus**
@BoltzmannBrain 

This covers both IMBU-142 and IMBU-143, although, the tests are failing for me due to an unrelated nupic failure (see my comment at https://github.com/numenta/nupic/pull/3042/files#r55761755).

### CODE REVIEW

**Comment by BoltzmannBrain** [comment]
A couple comments, otherwise :+1: 
I think we should wait on @subutai to respond to [your comment here](https://github.com/numenta/nupic/pull/3042/files#r55761755) so the tests pass before we merge.

**Comment by oxtopus** [comment]
I've updated the pipeline and configured it to test these changes at https://ci.numenta.com/browse/IMBU-IMBU2

File: projects/imbu/README.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+Additionally, there are some optional `docker run` arguments:
+
+- Map the `/opt/numenta/imbu/cache` directory in the container to one on on the
+  host external to the container to persist model state across containers.  For
+  example:
+
+  ```
+  -v `pwd`/cache:/opt/numenta/imbu/cache
+  ```
+- You may use a local clone of nupic.research in the container by mapping the
```
Comment (lscheinkman) [inline projects/imbu/README.md]: @oxtopus Is this still applicable?

**Comment by lscheinkman** [comment]
@oxtopus I got the following error with a clean docker image:
```
POST /fluent/CioDocumentFingerprint => generated 43243 bytes in 28 msecs (HTTP/1.1 500) 7 headers in 319 bytes (2 switches on core 0)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/web.py-0.37-py2.7.egg/web/application.py", line 239, in process
    return self.handle()
  File "/usr/local/lib/python2.7/dist-packages/web.py-0.37-py2.7.egg/web/application.py", line 230, in handle
    return self._delegate(fn, self.fvars, args)
  File "/usr/local/lib/python2.7/dist-packages/web.py-0.37-py2.7.egg/web/application.py", line 420, in _delegate
    return handle_class(cls)
  File "/usr/local/lib/python2.7/dist-packages/web.py-0.37-py2.7.egg/web/application.py", line 396, in handle_class
    return tocall(*args)
  File "engine/fluent_api.py", line 151, in POST
    response = g_fluent.query(modelName, data)
  File "engine/fluent_api.py", line 105, in query
    g_models[model] = imbu.createModel(model, str(loadPath), None)
  File "/opt/numenta/nupic.research/htmresearch/frameworks/nlp/imbu.py", line 250, in createModel
    raise ImbuUnableToLoadModelError(exc)
ImbuUnableToLoadModelError
```

**Comment by subutai** [comment]
@lscheinkman Please see https://github.com/numenta/nupic/pull/3050 - there's a backward compatibility error introduced in (my) PR from yesterday.

**Comment by oxtopus** [comment]
You may also need to update IMBU_LOAD_DATA_PREFIX envvar to /opt/numenta/nupic.research/projects/imbu/engine

File: projects/imbu/README.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+`docker run` command specified in the `IMBU_DOCKER_EXTRAS` environment
+variable.  For example, to map the directories described above.  It can be
+helpful to define all relevant environment variables in a file that you can
+either `source` at-will, or in your bash profile.  For example:
+
+```
+export IMBU_DOCKER_EXTRAS="-v <path to cache>:/opt/numenta/imbu/cache -v <path to nupic.research>:/opt/numenta/nupic.research"
+export CORTICAL_API_KEY="<cortical.io api key>"
+export IMBU_RETINA_ID="en_associative"
+export IMBU_LOAD_PATH_PREFIX="/opt/numenta/imbu/engine"
```
Comment (lscheinkman) [inline projects/imbu/README.md]: Based on https://github.com/numenta/nupic.research/pull/482#issuecomment-195107762 this variable should now be `/opt/numenta/nupic.research/projects/imbu/engine`

**Comment by oxtopus** [comment]
Thanks, @lscheinkman @BoltzmannBrain and @subutai.  I've fixed the README/setup.py per comments, and @subutai's fix has been merged.

**Pull request merged**